src/AzureUtils/Utils/BlobUtils.cs

```diff
@@ -10,38 +12,51 @@ namespace Etg.Yams.Azure.Utils
 {
     public static class BlobUtils
     {
-        public static Task DownloadBlobDirectory(CloudBlobDirectory blobDirectory, string destPath)
+        public static async Task DownloadBlobDirectory(CloudBlobDirectory blobDirectory, string destPath)
         {
-            return DownloadBlobs((dynamic) blobDirectory, destPath);
-        }
-
-        public static Task DownloadBlobContainer(CloudBlobContainer blobContainer, string destPath)
-        {
-            return DownloadBlobs((dynamic) blobContainer, destPath);
-        }
-
-        private static async Task DownloadBlobs(dynamic blobDirectory, string destPath)
-        {
-            await FileUtils.DeleteDirectoryIfAny(destPath);
-            await FileUtils.CreateDirectory(destPath);
+            if (!Directory.Exists(destPath))
+            {
+                await FileUtils.CreateDirectory(destPath);
+            }
 
             IEnumerable<IListBlobItem> blobs = await ListBlobsFlat(blobDirectory);
             var tasks = new List<Task>();
-            foreach (var blobItem in blobs)
+
+            using (var md5 = new MD5CryptoServiceProvider())
             {
-                var blob = (CloudBlockBlob) blobItem;
-                await blob.FetchAttributesAsync();
-                var localPathBlob = Path.Combine(destPath, GetLocalRelativePath(blob, blobDirectory));
-                string dirPath = Path.GetDirectoryName(localPathBlob);
-                if (!Directory.Exists(dirPath))
+                foreach (var blobItem in blobs)
                 {
-                    Directory.CreateDirectory(dirPath);
+                    var blob = (CloudBlockBlob)blobItem;
+                    await blob.FetchAttributesAsync();
+                    string relativePath = GetLocalRelativePath(blob, blobDirectory);
+                    var localFilePath = Path.Combine(destPath, relativePath);
+                    string dirPath = Path.GetDirectoryName(localFilePath);
+                    if (!Directory.Exists(dirPath))
+                    {
+                        Directory.CreateDirectory(dirPath);
```

Comment:
Why not using `FileUtils.CreateDirectory`?

Reply:
No good reason :) Will fix.